frontend/src/components/ShootWorkers/WorkerGroup.vue

```diff
@@ -65,81 +67,112 @@ export default {
+
+      if (minimum && maximum) {
+        description.push({
+          title: 'Autoscaler',
+          value: `Min. ${minimum} / Max. ${maximum}`
+        })
+      }
+      if (maxSurge) {
+        description.push({
+          title: 'Max. Surge',
+          value: `${maxSurge}`
+        })
+      }
+      if (zones.length) {
+        description.push({
+          title: zones.length > 1 ? 'Zones' : 'Zone',
+          value: join(zones, ', ')
+        })
+      }
+
+      return compact(description)
+    }
+  },
+  methods: {
+    machineTpeDescription (machine, machineType) {
+      if (machineType) {
+        return {
+          title: 'Machine Type',
+          value: machineType.name,
+          description: `(CPU: ${machineType.cpu} | GPU: ${machineType.gpu} | Memory: ${machineType.memory})`
         }
       }
-      if (this.workerGroup.volume) {
-        const volumeType = find(this.volumeTypes, { name: this.workerGroup.volume.type })
+      return {
+        title: 'Machine Type',
+        value: machine.type
+      }
+    },
+    volumeTpeDescription (machine, machineType, volume, volumeType) {
+      if (volume) {
         if (volumeType) {
-          description.push({
+          return {
             title: 'Volume Type',
-            value: `${volumeType.name} / ${this.workerGroup.volume.size}`,
+            value: volumeType.name,
             description: `(Class: ${volumeType.class})`
-          })
-        } else {
-          description.push({
-            title: 'Volume Type',
-            value: this.workerGroup.volume.type
-          })
+          }
+        }
+        return {
+          title: 'Volume Type',
+          value: volume.type
         }
       }
-      if (this.workerGroup.machine.image) {
-        const machineImage = find(this.machineImages, this.workerGroup.machine.image)
 
-        if (machineImage) {
-          const machineImageDescription = {
-            title: 'Machine Image',
-            value: `${machineImage.name} | Version: ${machineImage.version}`
-          }
-          if (machineImage.expirationDate) {
-            machineImageDescription.description = `(Expires: ${machineImage.expirationDateString})`
-          }
-          description.push(machineImageDescription)
-        } else {
-          description.push({
-            title: 'Machine Image',
-            value: this.workerGroup.machine.image
-          })
+      const storageType = get(machineType, 'storage.type')
+      if (storageType) {
+        return {
+          title: 'Volume Type',
+          value: storageType,
+          description: `(Class: ${machineType.storage.class})`
         }
       }
-      if (this.workerGroup.minimum && this.workerGroup.maximum) {
-        description.push({
-          title: 'Autoscaler',
-          value: `Min. ${this.workerGroup.minimum} / Max. ${this.workerGroup.maximum}`
-        })
+    },
+    volumeSizeDescription (machine, machineType, volume) {
```

Comment:
I assume the leading information is the `volume` and the fallback is the machineType. Maybe even optional. This should be reflected in the function parameters